validator/src/main.rs

```diff
@@ -2075,6 +2104,23 @@ pub fn main() {
     let account_indexes = process_account_indexes(&matches);
 
     let restricted_repair_only_mode = matches.is_present("restricted_repair_only_mode");
+    let accounts_shrink_optimize_total_space =
+        value_t_or_exit!(matches, "accounts_shrink_optimize_total_space", bool);
+    let shrink_ratio = value_t_or_exit!(matches, "accounts_shrink_ratio", f64);
+    if !(0.0..=1.0).contains(&shrink_ratio) {
```

Comment:
Can you set arg value limits that automatically check bounds for you? .takes_value, etc. I think there are range-limiting functions. I could be wrong.

Reply:
I wondered about the same and I did not find anything like that. I think the Arg does not have a mechanism to set the types of the argument value and that is probably we use value_t_or_exit! later to check the types.